chore: Add component owners workflow, documentation

kaylareopelle opened the pull request:
Add the component owners workflow to automatically assign owners to pull requests when changes are proposed for their gem.

~~This is blocked until we can make sure all component owners are members of the OpenTelemetry organization on GitHub and are added as members of the ruby-contrib-triagers team.~~ _This shouldn't be a blocker, the JS team had names on the list that weren't members while still having the workflow setup_

kaylareopelle committed:
chore: Add component owners workflow

Add the component owners workflow to automatically assign owners to
pull requests when changes are proposed for their gem.

This is blocked until we can make sure all component owners are members
of the OpenTelemetry organization on GitHub and are added as members of
the ruby-contrib-triagers team.

kaylareopelle committed:
ci: Add Triager, triagers to known words

kaylareopelle committed:
Merge branch 'main' into component-owners

arielvalentin commented on .github/workflows/component-owners.yml:
```diff
@@ -0,0 +1,18 @@
+      - uses: dyladan/component-owners@v0.1.0
+        with:
+          config-file: .github/component_owners.yml
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
```
```suggestion
          repo-token: ${{ secrets.GITHUB_TOKEN }}

```

kaylareopelle committed:
Apply suggestions from code review

Co-authored-by: Ariel Valentin <arielvalentin@users.noreply.github.com>

kaylareopelle commented on CODEOWNERS:
```diff
@@ -1,41 +0,0 @@
-#  https://help.github.com/en/articles/about-code-owners
-#
-
-* @open-telemetry/ruby-contrib-maintainers @open-telemetry/ruby-contrib-approvers @fbogsany @mwear @robertlaurin @dazuma @ericmustin @arielvalentin @ahayworth @plantfansam @robbkidd @simi @kaylareopelle @xuan-cao-swi
```
CNCF Slack Thread for these questions: https://cloud-native.slack.com/archives/C01NJ7V1KRC/p1761948733868689

kaylareopelle committed:
Merge branch 'main' into component-owners

kaylareopelle committed:
Add factory_bot to component owners

kaylareopelle committed:
Keep wildcard in codeowners

This intends to continue our practice of having every maintainer
and approver directly assigned to
PRs

kaylareopelle committed:
Merge branch 'component-owners' of https://github.com/kaylareopelle/opentelemetry-ruby-contrib into component-owners

arielvalentin reviewed (commented):
Do you know if the component owners action fails if a new component is added but is not declared?

Also. I would like to have the instrumentation generator update the component owners file, that way when someone submits an instrumentation then they will declare themselves as component owners: 

https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/.instrumentation_generator/instrumentation_generator.rb

arielvalentin committed:
Update CODEOWNERS

arielvalentin committed:
Merge branch 'main' into component-owners

kaylareopelle committed:
Merge branch 'main' into component-owners

kaylareopelle commented:
> Do you know if the component owners action fails if a new component is added but is not declared?
> 
> Also. I would like to have the instrumentation generator update the component owners file, that way when someone submits an instrumentation then they will declare themselves as component owners:
> 
> https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/.instrumentation_generator/instrumentation_generator.rb

I'm sorry, @arielvalentin, I missed this before I merged. Here's a new PR to address add component owners to the generator, along with a few other fixes: https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/1792
